src/multipart/multipart.rs

```diff
@@ -0,0 +1,469 @@
+extern crate uuid;
+
+use std;
+use std::borrow::Cow;
+use std::io::Read;
+use std::ascii::AsciiExt;
+use hyper::mime::Mime;
+
+/// A multipart/form-data request.
+#[derive(Debug)]
+pub struct MultipartRequest {
+    boundary: String,
+    fields: Vec<MultipartField>,
+}
+
+impl MultipartRequest {
+    /// Creates a new MultipartRequest without any content.
+    pub fn new() -> MultipartRequest {
+        MultipartRequest {
+            boundary: format!("{}", uuid::Uuid::new_v4().simple()),
```

Comment:
I wonder if users would want to set their own boundary... Looking around in python requests and node request, it doesn't seem like they provide an option. I suppose if it is needed, it can be added later.